pkg/limayaml/validate.go

```diff
@@ -385,6 +386,46 @@ func validateNetwork(y *LimaYAML) error {
 	return nil
 }
 
+// ValidateParamIsUsed checks if the keys in the `param` field are used in any script, probe, copyToHost, or portForward.
+// It should be called before the `y` parameter is passed to FillDefault() that execute template.
+func ValidateParamIsUsed(y *LimaYAML) error {
+	for key := range y.Param {
+		re, err := regexp.Compile(`{{[^}]*\.Param\.` + key + `[^}]*}}`)
```

Comment:
This expression doesn't match when the parameter is only used in a template expression like this:

```go
"{{if eq .Param.Rootful \"true\"}}/var/run{{else}}/run/user/{{.UID}}{{end}}/docker.sock"
```

If would suggest to loosen up the regex, even if it could potentially get some false positives. That is still better than missing references and throwing an error:

```suggestion
		re, err := regexp.Compile(`\.Param\.` + key + `\b`)
```

Also add a test case for this scenario.

Reply:
I added a test using `"{{if eq .Param.rootful \"true\"}}/var/run{{else}}/run/user/{{.UID}}{{end}}/docker.sock"` to `TestValidateParamIsUsed`. It detected the pattern without needing to change the pattern passed to regexp, so that part remains unchanged.